### ISSUE TITLE
Have you gotten this to work with a frida-gadget?

### ISSUE THREAD

**Issue by rdbo** [commonuserlol/frida-java-menu#7]
This is not so much of a bug, the menu works really well, it's more of a question. Have you gotten this to work with a frida-gadget (load script)?
I have tried embedding the Example Menu from the Wiki into an APK with frida-gadget using Objection, but I couldn't get it working. If I just run it normally, with frida-server without embedding the script in the APK, it works fine.
Steps I did:
- Decompiled the APK
- Added permission for Display over other Apps (`SYSTEM_ALERT_WINDOW`)
- Recompiled the APK
- Ran `objection` with a config that will load my `agent.js` with the frida gadget
I've had problems with other menus as well, I'm just wondering if I'm missing something.
I'm using Android 13.0 (Lineage OS), frida 16.1.4 (latest), objection 1.11.0 (latest)

**Comment by commonuserlol** [comment]
Nope I didn't since it just not loading.. No idea what's wrong but try look to logcat. Possible solutions is:

- Run `adb root` (lineage allows this) then `adb shell pm path com.asd` and check is libs correctly copied (`adb shell ls (path which got by pm path)/libs/arch/`, if needed pull them). But is with other scripts it works? I mean try load menu via gadget (script) and then other script which umm just toast or smth.
- Use other tool for patch
- Do manual patching (google "frida android no root")

I will be afk for some days so can't respond fast or even won't respond

**Comment by commonuserlol** [comment]
By other script I mean patch apk with (1) script with menu, (2) other script
But you should try other tools since objection kinda dead since 2021 and maybe something wrong

**Comment by commonuserlol** [comment]
A12/12.1 same behavior, later I will try with a6, a9

**Comment by rdbo** [comment]
By the way, from my testing, it seems that `Menu.MainActivity.waitForInit(main)` isn't being called at all (even though the code is there). I tried this:
```ts
Menu.MainActivity.waitForInit(main).then(() => {
  Log.d("[TEST MOD]", "waitForInit finished");
});
```
Guess what? No log shown in `adb logcat` about `waitForInit finished`. Maybe a problem with frida itself?

**Comment by rdbo** [comment]
Fixed by #8
